test/python/circuit/test_gate_definitions.py

```diff
@@ -113,6 +115,42 @@ def test_cx_definition(self):
         decomposed_circ = circ.decompose()
         self.assertTrue(Operator(circ).equiv(Operator(decomposed_circ)))
 
+class TestGateEquivalenceEqual(QiskitTestCase):
+    """Test the decomposition of a gate in terms of other gates
+    yields the same matrix as the hardcoded matrix definition."""
+
+    @classmethod
+    def setUpClass(cls):
+        class_list = Gate.__subclasses__() + ControlledGate.__subclasses__()
+        exclude = {'ControlledGate', 'DiagonalGate', 'UCGate', 'MCGupDiag',
+                   'MCU1Gate', 'UnitaryGate', 'HamiltonianGate',
+                   'UCPauliRotGate', 'SingleQubitUnitary', 'MCXGate',
+                   'VariadicZeroParamGate'}
+        cls._gate_classes = []
+        for aclass in class_list:
+            if aclass.__name__ not in exclude:
+                cls._gate_classes.append(aclass)
+
+    def test_equivalence_phase(self):
+        """Test that the equivalent circuits from the equivalency_library
+        have equal matrix representations"""
+        for gate_class in self._gate_classes:
+            with self.subTest(i=gate_class):
```

Comment:
```suggestion
    @data(self._gate_classes)
    def test_equivalence_phase(self):
        """Test that the equivalent circuits from the equivalency_library
        have equal matrix representations"""

```

Reply:
I found this style useful for debugging but can use this suggestion.

Reply:
I wasn't able to reference self._gate_classes in the call to @data.